File: flip_bits.cpp
#include <bits/stdc++.h>
using namespace std;

void invertBits(int num)
{
    int x = log2(num) + 1;

    for (int i = 0; i < x; i++)
       num = (num ^ (1 << i));

    cout << num;
}

// Driver code
int main()
{
    int num = 26;
    invertBits(num);
    return 0;
}

File: Intermediate/flip_bits.py
import math 
  
def bitsflipper(num): 
    len = int(math.log2(num)) + 1
    for i in range(len):  
        num = (num ^ (1 << i))
    return num

n = int(input("Enter a number: "))
print("The number obtained after flipping the bits is:",bitsflipper(n))
